packages/source-iottwinmaker/src/property-value/client/getPropertyValueByEntity.ts

```diff
@@ -0,0 +1,60 @@
+import { GetPropertyValueCommand, GetPropertyValueResponse, IoTTwinMakerClient } from '@aws-sdk/client-iottwinmaker';
+import { DataType, DataStreamBase } from '@iot-app-kit/core';
+import { TwinMakerStaticDataQuery } from '../types';
+import { toValue } from '../../utils/propertyValueUtils';
+import { isNumber } from 'lodash';
+
+export const getPropertyValueByEntity = async ({
+  client,
+  query,
+}: {
+  client: IoTTwinMakerClient;
+  query: TwinMakerStaticDataQuery;
+}): Promise<DataStreamBase[]> => {
+  const results: DataStreamBase[] = [];
+  let nextToken: string | undefined = undefined;
+  do {
```

Comment:
Maybe you can add a try catch block here and handle potential failures?

Reply:
the failures will be thrown to caller and caught there.